index.js

```diff
@@ -13,7 +13,19 @@ function BroccoliMergeTrees(inputNodes, options) {
     throw new Error('Expected array, got ' + inputNodes)
   }
   options = options || {}
-  Plugin.call(this, inputNodes, {
+  var newInputNodes = []
+  for (var i = 0 ; i < inputNodes.length ; i++ ) {
+    if (inputNodes[i].constructor.name  === this.constructor.name ) {
```

Comment:
this comparison seems somewhat dubious, i wish we had better dedupe rules in node, then just straight constructor equality would suffice.

It would be nice if some reasonable protocol existed here.

@joliss anything come to mind?